### DISCUSSION TITLE
Automatically Segment and Label Objects in Video (Project 203)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#33] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project #203](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/tree/main/projects/Automatically%20Segment%20and%20Label%20Objects%20in%20Video)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by cyl0000** [comment]
Is this project to track multiple objects, such as all the vehicles in the scene, or to track a single object, such as a static or dynamic specific object?

**Reply by mwbpatel** [reply]
Hi,

The main goal of the project is to develop a label automation algorithm that can generate pixel level labels for a single object (dynamic or static) across multiple video frames. 

The automation algorithm should make it easier for a user to generate pixel level labels without a human user having to label each individual video frame. 

Here is one example of what such an automation algorithm might look like:
https://openaccess.thecvf.com/content/CVPR2021/papers/Cheng_Modular_Interactive_Video_Object_Segmentation_Interaction-to-Mask_Propagation_and_Difference-Aware_Fusion_CVPR_2021_paper.pdf

You can search for other research papers using these search terms: "video object segmentation" or "video instance segmentation".

Thanks,
Birju

**Comment by cyl0000** [comment]
We have some questions about this project:
1. Can you provide some data set?
2. Is there any MATLAB algorithm for reference?
3. Do we need to mark a vehicle or divide the ground, trees and cars in the whole video by semantic segmentation like this:https://ww2.mathworks.cn/help/vision/ug/getting-started-with-semantic-segmentation-using-deep-learning.html
4.Do we need to track a specific object?
5. Is it only necessary to perform instance segmentation in each frame of the video, similar to YOLOv4?

**Reply by mwbpatel** [reply]
Hi,

I would recommend using datasets that are publicly available for academic or research purposes. You can have a look at this dataset to get started: 

https://davischallenge.org/

https://jonathonluiten.medium.com/robmots-the-ultimate-tracking-challenge-at-cvpr21-6fb3d11ab025

There are additional datasets that you can search for online as well. 

In terms of MATLAB algorithms, we have several semantic segmentation networks (e.g. U-Net) and Mask R-CNN for instance segmentation that you can use to get started. However, the project will require to build something that we don't have a reference algorithm for at this time. For example, you may start with instance segmentation to identify the object to label, but then you have to come w/ an algorithm to automatically label subsequent instances of that object in the next video frames. You may need to use tracking to do this, but it depends on what you decide to implement. 

It would be good to think about algorithms that leverage temporal information in the video to help propagate the labels to subsequent frames versus using something like YOLO v4 which does not leverage any temporal information. 

Thanks,
Birju

**Comment by cyl0000** [comment]
Is it only necessary to mark one kind of things , like the vehicle? Or do the vehicles, pedestrians, cyclists, lane markers, drivable path, curbs and walkways in each frame of the video need to be split?

**Reply by mwbpatel** [reply]
I would try to get one kind of thing working before you try to do multiple objects.

**Comment by KAZUKI899** [comment]
Our group feels that our current algorithm is more focused on object detection and tracking, and we don't know whether to add it to the video label。If you want to add, there are no teaching materials。

**Comment by KAZUKI899** [comment]
Our group feels that our current algorithm is more focused on object detection and tracking, and we don't know whether to add it to the video label。If you have teaching materials,could you send us?

**Reply by mwbpatel** [reply]
You should use the current algorithm as an example of how to implement a custom automation algorithm. The goal of the project to write a new automation algorithm for segmenting objects across video frames. 

The background materials section on the project page contains links to documentation that teaches you about the automation algorithm API and has links to research papers outlining various algorithms and techniques for segmenting objects across video frames.

**Comment by 004Ajay** [comment]
Can I do this project in Python and still submit it to Matlab?

**Reply by robertogl** [reply]
When you say "submit it to MATLAB" you mean submit your Python solution to us via the submission form?
In that case, you will not be eligible for the rewards.
